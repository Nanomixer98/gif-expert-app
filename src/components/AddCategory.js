import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); // undefined
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2) {
            setCategories((cats) => [...cats, e.target[0].value])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.prototypes = {
    setCategories: PropTypes.func.isRequired
}