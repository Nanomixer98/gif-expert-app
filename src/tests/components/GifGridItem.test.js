import "@testing-library/jest-dom";
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem";

describe('Prueba en <GifGridItem />', () =>{

    const title = 'Un título';
    const url = 'https://localhost/algo.jpg'
    let wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Debe mostrar correctamente <GifGridItem />', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un párrafor con el title', () => { 
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Debe de tener la imagen igual al url y alt de los props', () => { 
        const img = wrapper.find('img');
        // console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe tener animate__fadeIn', () => { 
        const div = wrapper.find('div');
        // console.log(div.hasClass('animate__fadeIn1'));
        expect(div.hasClass('animate__fadeIn')).toBe(true);
    });

});