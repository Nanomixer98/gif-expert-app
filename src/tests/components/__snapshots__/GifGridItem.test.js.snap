// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> Debe mostrar correctamente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__rollIn animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un título
  </p>
</div>
`;
